pandas/tests/indexes/test_base.py

```diff
@@ -744,19 +736,31 @@ def test_intersect_str_dates(self):
         assert len(res) == 0
 
     def test_union(self):
+        # TODO: Replace with fixtures
         first = self.strIndex[5:20]
         second = self.strIndex[:10]
         everything = self.strIndex[:20]
+
         union = first.union(second)
         assert tm.equalContents(union, everything)
 
+    @pytest.mark.parametrize("klass", [
```

Comment:
this notion of klass we have all over the place, sometimes its also called box (not necessary to do here, just letting you know, prob best to update the issue you created)